jobs/get_all_cards_by_date_range_gold.py

```diff
@@ -0,0 +1,89 @@
+"""
+This script will fetch all the cards for the given user according to the below criteria:
+- get the date within the given date range 
+- save the data to Delta Lake Gold table
+- This script is scheduled to run twice a week
+"""
+
+import os
+import sys
+from datetime import datetime, timedelta
+import argparse
+from loguru import logger
+from pyspark.sql.functions import col, from_utc_timestamp, to_date
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from helpers import spark_helper as sh, common_functions as cf, trello_functions as tf
+from config import delta_table_conf as dc
+
+def get_user_cards(spark, start_date, end_date):
+    """
+    Fetch all the cards data for the given user within the given date range 
+    and save it into a Delta Lake Gold table
+    :param spark: Spark session
+    :param start_date: start date
+    :param end_date: end date
+    :return none
+    """
+    path = dc.indira_cards_gold_path
+    try:
+        # Fetch the cards data from the refined cards silver delta table
+        refined_cards_data = spark.read.format(
+            "delta").load(dc.cards_unique_silver_table_path)
+        logger.info(
+            f'Fetched refined card data from silver delta table')      
+        
+        # Convert from_date to a datetime object
+        start_date = datetime.strptime(start_date, '%Y-%m-%d')
+
+        # Convert "dateLastActivity" column to date format
+        refined_cards_data = refined_cards_data.withColumn("dateLastActivity", to_date(from_utc_timestamp("dateLastActivity", "UTC")))
```

Comment:
@indiranell Ran the script and it was successful. However, if using UTC, it is removing the timestamp too. And though the comparison works, it is better to have the timestamp too. So, instead can you do :
start_date = datetime.strptime(start_date, "%Y-%m-%d")

Reply:
Done. Removed the UTC related code

Reply:
Thank you Indira. After I put this comment, I realized it is actually fine to have UTC. It is the way you have designed your search. So, I am sorry for this comment. Do you want to revert back to the previous or keep this? Either way should be fine.

Reply:
@sravantit25 Lets keep it like this for now.

Reply:
Sure, Indira. Thanks! All looks good, merging the PR.